hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheInvalidatorImpl.java

```diff
@@ -146,7 +143,7 @@ private int getBackgroundProcessorRunPeriodSeconds() {
 
     @Override
     public void invalidateLocalNearCache(String mapName, Data key) {
-        if (!isNearCacheAndInvalidationEnabled(mapName)) {
```

Comment:
It would be better, if you use `MapContainer` inside class that calls `NearCacheInvalidatorImpl` (I don't know IMap implementation but it might be `AbstractRecordStore` or `DefaultRecordStore`) to check if there is any registered listener or not. So there wouldn't be `mapServiceContext.getOrNullMapContainer(mapName);` every time to fire event to check is there any registered listener or not.